src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -89,7 +97,18 @@ uiModules.get('apps/management')
 
   const updateFieldList = results => {
     this.fetchFieldsError = results.fetchFieldsError;
-    this.dateFields = results.dateFields;
+    if (this.fetchFieldsError) {
+      return;
+    }
+
+    this.dateFields = results.dateFields || [];
+    this.doesIndexHaveDateFields = this.dateFields.length > 0;
```

Comment:
`this.indexHasDateFields` fits the pattern better, and reads better when used for conditionals.